### PR TITLE
Add `aisuite` support

### PR DESCRIPTION
**Pull request by davidberenstein1957**
- support for `aisuite`
- basic integration tests based on `examples/observers` execution. We can consider extending this to `stores` but this would require a bit more mocking efforts.
- code cleaning

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -43,7 +49,7 @@ response = client.chat.completions.create(
 ### Supported Observers
 
 - [OpenAI](https://openai.com/) and every other LLM provider that implements the [OpenAI API message formate](https://platform.openai.com/docs/api-reference)
-
+- [AISuite](https://github.com/aisuite-ai/aisuite)
```
Comment (cfahlgren1) [inline README.md:52]:
```suggestion
- [AISuite](https://github.com/andrewyng/aisuite)
```

**Comment by cfahlgren1** [comment]
Pushed a tiny commit to fix comment for an example

File: examples/observers/aisuite_example.py [hunk 2]
```diff
@@ -0,0 +1,19 @@
+import aisuite as ai
+from observers.observers import wrap_aisuite
+
+client = ai.Client()
+
+client = wrap_aisuite(client)
+
+models = ["openai:gpt-4o", "anthropic:claude-3-5-sonnet-20240620"]
```
Comment (cfahlgren1) [inline examples/observers/aisuite_example.py:8]:
```suggestion
# ensure you have both `ANTHROPIC_API_KEY` and `OPENAI_API_KEY` environment variables set
models = ["openai:gpt-4o", "anthropic:claude-3-5-sonnet-20240620"]
```

**Review by cfahlgren1 — APPROVED** [review]
Great job! Looks good.

image

**Comment by cfahlgren1** [comment]
I wonder since aisuite implements openai if we don't necessarily need this and_ can just show an example with 

`wrap_openai` on aisuite client?

unless you think the alias is useful?

**Comment by davidberenstein1957** [comment]
@cfahlgren1 I assume it still routes the logic to other LLM providers and does some of the mappings to the OpenAI schema for us, so I would say it is nice to add. Similarly to `litellm` also, just to have the coverage in case people are already using either one of the frameworks.

**Pull request merged**